packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Writers/CodeWriterDeclarationTests.cs

```diff
@@ -163,7 +163,7 @@ public void ScopeDeclaredTwiceForMethodSignatureParam()
 
             using var codeWriter = new CodeWriter();
             codeWriter.WriteMethod(method);
-            var declScopes = GetDeclarationScopes(param.AsExpression.Declaration);
+            var declScopes = GetDeclarationScopes(((VariableExpression)param).Declaration);
```

Comment:
I thought we were going to replace this with an extension method call on ParameterProvider? Using a cast here makes the code harder to reason about than the previous code.

Reply:
updated